var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -108,125 +218,244 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
+            "raja +cuda cuda_arch={0}".format(sm_),
+            when="raja_offload=nvidia cuda_arch={0}".format(sm_),
+        )
+    depends_on("kokkos +openmp", when="kokkos_backend=omp")
+    depends_on("raja +openmp", when="raja_offload=cpu")
 
     # OpenCL Dependency
-
-    backends = {
-        "ocl": [
-            ("amd", "rocm-opencl", "enable ROCM backend"),
-            ("cuda", "cuda", "enable Cuda backend"),
-            ("intel", "intel-oneapi-compilers", "enable Intel backend"),
-            ("pocl", "pocl@1.5", "enable POCL backend"),
-        ],
-        "kokkos": [
-            ("cuda", "cuda", "enable Cuda backend"),
-            ("omp", "none", "enable Cuda backend"),
-        ],
-    }
-    backend_vals = ["none"]
-    for lang in backends:
-        for item in backends[lang]:
-            backend, dpdncy, descr = item
-            backend_vals.append(backend.lower())
-
-    variant("backend", values=backend_vals, default="none", description="Enable backend support")
-
-    for lang in backends:
-        for item in backends[lang]:
-            backend, dpdncy, descr = item
-            if dpdncy.lower() != "none":
-                depends_on("%s" % dpdncy.lower(), when="backend=%s" % backend.lower())
-    # this flag could be used in all required languages
-    variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
-
-    # comp_impl_vals=["ONEAPI-DPCPP","DPCPP","HIPSYCL","COMPUTECPP"]
     variant(
-        "implementation",
-        values=str,
+        "ocl_backend",
+        values=("amd", "cuda", "intel", "pocl", "none"),
         default="none",
-        description="Compile using the specified SYCL compiler option",
+        when="+ocl",
+        description="Enable Backend Target for OpenCL",
     )
-
-    conflicts(
-        "implementation=none",
-        when="+sycl",
-        msg="SYCL requires compiler implementation to be specified by option=",
+    variant(
+        "kokkos_backend",
+        values=("cuda", "omp", "none"),
+        default="none",
+        when="+kokkos",
+        description="Enable Backend Target for kokkos",
     )
     conflicts(
-        "implementation=none",
-        when="+thrust",
-        msg="Which Thrust implementation to use, supported options include:\
-         - CUDA (via https://github.com/NVIDIA/thrust)\
-         - ROCM (via https://github.com/ROCm/rocThrust)",
+        "ocl_backend=none",
+        when="+ocl",
+        msg="OpenCL implementation requires backend to be specfied by ocl_backend=",
     )
+    # depends_on("rocm-opencl@6.0.2", when="+ocl ocl_backend=amd")
+    depends_on("cuda", when="+ocl ocl_backend=cuda")
+    depends_on("cuda", when="+sycl2020 sycl2020_offload=nvidia")
+    depends_on("intel-oneapi-compilers", when="+ocl ocl_backend=intel")
+    depends_on("pocl@1.5", when="+ocl ocl_backend=pocl")
+
+    variant("flags", values=str, default="none", description="Additional CXX flags to be provided")
```

Comment:
Actually, great attention to detail, it should be named as `cuda_flags` since we are using it for providing extra cuda flags if needed adding it to next commit